backend/respApi/restApi.go

```diff
@@ -0,0 +1,160 @@
+package respApi
+
+import (
+	. "backend/extensionController"
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/exasol/exasol-driver-go"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"sync"
+)
+
+type RestApi struct {
+	Controller   ExtensionController
+	server       *http.Server
+	stopped      *bool
+	stoppedMutex *sync.Mutex
+}
+
+func (restApi *RestApi) Serve() {
+	if restApi.server != nil {
+		panic("server already running")
+	}
+	restApi.setStopped(false)
+	router := gin.Default()
+	router.GET("/extensions", restApi.handleGetExtensions)
+	router.GET("/installations", restApi.handleGetInstallations)
+	srv := &http.Server{
+		Addr:    ":8080",
+		Handler: router,
+	}
+	restApi.server = srv
+	err := restApi.server.ListenAndServe() // blocking
+	if err != nil && !restApi.isStopped() {
+		panic(fmt.Sprintf("failed to start rest API server. Cause: %v", err.Error()))
+	}
+}
+
+func (restApi *RestApi) setStopped(stopped bool) {
+	if restApi.stopped == nil {
+		stopped := false
+		restApi.stopped = &stopped
+		restApi.stoppedMutex = &sync.Mutex{}
+	}
+	restApi.stoppedMutex.Lock()
+	defer restApi.stoppedMutex.Unlock()
+	*restApi.stopped = stopped
+}
+
+func (restApi *RestApi) isStopped() bool {
+	restApi.stoppedMutex.Lock()
+	defer restApi.stoppedMutex.Unlock()
+	return *restApi.stopped
+}
+
+func (restApi *RestApi) Stop() {
+	if restApi.server == nil {
+		panic("cant stop server since it's not running")
+	}
+	restApi.setStopped(true)
+	err := restApi.server.Shutdown(context.Background())
+	if err != nil {
+		panic(fmt.Sprintf("failed to shutdown rest API server. Cause: %v", err.Error()))
+	}
+	restApi.server = nil
+}
+
+func (restApi *RestApi) handleGetExtensions(c *gin.Context) {
+	response, err := restApi.getExtensions(c)
+	restApi.sendResponse(c, response, err)
+}
+
+func (restApi *RestApi) getExtensions(c *gin.Context) (*ExtensionsResponse, error) {
+	dbConnection, err := restApi.openDbConnection(c)
+	if err != nil {
+		return nil, err
+	}
+	defer closeDbConnection(dbConnection)
```

Comment:
Another thing for a future change: add a DB connection pool :)

Reply:
It's tricky because the Saas backend hadles all customers DBs. So we need one ppol per customer. And we should make sure that it's not possible to take over the connection of another customer

Reply:
👍